app/protocol/flows/v3/register.go

```diff
@@ -49,6 +50,7 @@ func registerAddressFlows(m protocolManager, router *routerpkg.Router, isStoppin
 
 func registerBlockRelayFlows(m protocolManager, router *routerpkg.Router, isStopping *uint32, errChan chan error) []*common.Flow {
 	outgoingRoute := router.OutgoingRoute()
+	ibdChannel := make(chan *externalapi.DomainBlock)
```

Comment:
Instead of creating the channel here and passing it to both flows, please add it to FlowContext.

Reply:
But FlowContext is shared between all peers, The IBD channel is per peer

Reply:
Good point. Please add it to Peer instead

Reply:
Are you sure? it seems a bit out of context for the Peer struct

Reply:
It's about as in-context as the last ping time. Certainly not more out of context than the flow registration function